tests/test_proxy_functional.py

```diff
@@ -1,15 +1,84 @@
 import asyncio
+import logging
 import os
 import pathlib
 from unittest import mock
 
+import proxy
 import pytest
 from yarl import URL
 
 import aiohttp
 from aiohttp import web
 
 
+@pytest.fixture
+def secure_proxy_url(aiohttp_unused_port, tls_ca_certificate_pem_path, tls_ca_private_key_pem_path):
+    """Return the an instance of a running secure proxy."""
+    unused_port = str(aiohttp_unused_port())
+    address = "127.0.0.1"
+    input_args = [
+        "--num-workers",
+        "1",
+        "--hostname",
+        "127.0.0.1",
+        "--port",
+        unused_port,
+        "--cert-file",
+        tls_ca_certificate_pem_path,
+        "--key-file",
+        tls_ca_private_key_pem_path,
+    ]
+    proxy_url = URL.build(scheme="https", host=address, port=unused_port)
+    with proxy.Proxy(input_args=input_args):
+        yield proxy_url
+
+
+# @pytest.mark.xfail
+# @pytest.mark.parametrize("protocol", ["http", "http"])
+async def test_secure_proxy_http_absolute_path(
+    aiohttp_server,
+    ssl_ctx,
+    secure_proxy_url,
+    get_request,
+    # protocol
+) -> None:
+    """Test urls can be requested through a secure proxy."""
+    async def handler(request):
+        return web.Response(text="Test message")
+    app = web.Application()
+    app.router.add_route("GET", "/", handler)
+    server = await aiohttp_server(app)
+
+    # url = URL.build(scheme=protocol, host=server.host, port=server.port)
+    url = URL.build(scheme="http", host=server.host, port=server.port)
+    response = await get_request(url=url, proxy=secure_proxy_url, ssl=False)
+    assert response.status == 200
+
+
+# @pytest.mark.xfail
+# @pytest.mark.parametrize("protocol", ["http", "http"])
+async def test_secure_proxy_https_absolute_path(
```

Comment:
These are still separate because the ssl contexts are different, but I plan to deduplicate.